File: php/example_code/cognito/CreateUserPool.php
<?php
/**
 * Copyright 2010-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * This file is licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License. A copy of
 * the License is located at
 *
 * http://aws.amazon.com/apache2.0/
 *
 * This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
 * CONDITIONS OF ANY KIND, either express or implied. See the License for the
 * specific language governing permissions and limitations under the License.
 *
 *  ABOUT THIS PHP SAMPLE: This sample is part of the SDK for PHP Developer Guide topic at
 * https://docs.aws.amazon.com/sdk-for-php/v3/developer-guide/iam-examples-working-with-policies.html
 *
 */
// snippet-start:[cognito.php.user_pool.create_user_pool.complete]
// snippet-start:[cognito.php.user_pool.create_user_pool.import]

require 'vendor/autoload.php';

use Aws\CognitoIdentityProvider\CognitoIdentityProviderClient;
use Aws\Exception\AwsException;

// snippet-end:[cognito.php.user_pool.create_user_pool.import]

/**
 * Creates a new User Pool Client for your AWS account.
 *
 * This code expects that you have AWS credentials set up per:
 * https://docs.aws.amazon.com/sdk-for-php/v3/developer-guide/guide_credentials.html
 */

 // snippet-start:[cognito.php.user_pool.create_user_pool.main]
 
$client = new CognitoIdentityProviderClient([
    'profile' => 'default',
    'region' => 'us-east-2',
    'version' => '2016-04-18'
]);

$userPoolName = "PHP_SDK_test_user_pool";

try {
    $result = $client->createUserPool([
        'PoolName' => $userPoolName, 
    ]);
    echo $result["UserPool"]["Id"] . "\n";
    var_dump($result);
} catch (AwsException $e) {
    // output error message if fails
    echo $e->getMessage() . "\n";
    error_log($e->getMessage());
}

// snippet-end:[cognito.php.user_pool.create_user_pool.main]
// snippet-end:[cognito.php.user_pool.create_user_pool.complete]
// snippet-comment:[These are tags for the AWS doc team's sample catalog. Do not remove.]
// snippet-sourcedescription:[CreateUserPool.php demonstrates how to create a user pool for Cognito.]
// snippet-keyword:[PHP]
// snippet-keyword:[AWS SDK for PHP v3]
// snippet-keyword:[Code Sample]
// snippet-keyword:[Amazon Cognito]
// snippet-service:[cognito]
// snippet-sourcetype:[full-example]
// snippet-sourcedate:[2019-03-29]
// snippet-sourceauthor:[jschwarzwalder (AWS)]

File: .kotlin_alpha/services/cognito/src/main/kotlin/com/kotlin/cognito/CreateUserPool.kt
//snippet-sourcedescription:[CreateUserPool.kt demonstrates how to create a user pool for Amazon Cognito.]
//snippet-keyword:[AWS SDK for Kotlin]
//snippet-keyword:[Code Sample]
//snippet-service:[Amazon Cognito]
//snippet-sourcetype:[full-example]
//snippet-sourcedate:[08/01/2021]
//snippet-sourceauthor:[scmacdon-aws]

/*
   Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
   SPDX-License-Identifier: Apache-2.0
*/

package com.kotlin.cognito

//snippet-start:[cognito.kotlin.create_user_pool.import]
import aws.sdk.kotlin.services.cognitoidentityprovider.CognitoIdentityProviderClient
import aws.sdk.kotlin.services.cognitoidentityprovider.model.CreateUserPoolRequest
import aws.sdk.kotlin.services.cognitoidentity.model.CognitoIdentityException
import kotlin.system.exitProcess
//snippet-end:[cognito.kotlin.create_user_pool.import]

suspend fun main(args: Array<String>){

    val usage = """
        Usage: <userPoolName>
    
        Where:
            userPoolName - the ID given to your user pool.
        """

     if (args.size != 1) {
          println(usage)
          exitProcess(0)
      }

    val userPoolName = args[0]
    val cognitoClient = CognitoIdentityProviderClient { region = "us-east-1" }
    val userPoolId = createPool(cognitoClient,userPoolName)
    print("The new user pool Id is $userPoolId")
    cognitoClient.close()
}

//snippet-start:[cognito.kotlin.create_user_pool.main]
 suspend fun createPool(cognitoClient: CognitoIdentityProviderClient, userPoolName:String): String? {

        try {

            val createUserPoolRequest = CreateUserPoolRequest{
                this.poolName = userPoolName
            }

            val createUserPoolResponse = cognitoClient.createUserPool(createUserPoolRequest)
            return createUserPoolResponse.userPool?.id;

        } catch (ex: CognitoIdentityException) {
            println(ex.message)
            cognitoClient.close()
            exitProcess(0)
        }
        return ""
  }
//snippet-end:[cognito.kotlin.create_user_pool.main]